pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/SumAggregationFunction.java

```diff
@@ -102,10 +130,112 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
     aggregationResultHolder.setValue(sum);
   }
 
+  private void aggregateNullHandlingEnabled(int length, AggregationResultHolder aggregationResultHolder,
+      BlockValSet blockValSet, RoaringBitmap nullBitmap) {
+    double sum = 0;
+    switch (blockValSet.getValueType().getStoredType()) {
+      case INT: {
+        int[] values = blockValSet.getIntValuesSV();
+        if (nullBitmap.getCardinality() < Math.min(length, values.length)) {
```

Comment:
Move value read into the if block to avoid unnecessary value read. Same for other places

Reply:
I did not move before because I was not sure if we need to do the check for both length and values.length. Moved.